airbyte-cdk/python/airbyte_cdk/sources/declarative/transformations/transformation.py

```diff
@@ -18,11 +18,11 @@ class RecordTransformation:
     @abstractmethod
     def transform(
         self,
-        record: Record,
+        record: Mapping[str, Any],
```

Comment:
Are we concerned about anyone inheriting from `RecordTransformation` and implementing this method with the assumption that they'll get a `Record` (i.e. that they'll have access to the `data` and `associated_slice` methods)?